### PR TITLE
Add address book plugins

### PR DESCRIPTION
**Pull request by ChristophWurst**
Somehow Github refused to reopen https://github.com/nextcloud/server/pull/19680, thus a new PR.

Same same as https://github.com/nextcloud/server/pull/19196 but different.

- [x] Pluggable address books
- [x] Load the address book for the user

Tested with https://github.com/nextcloud/server/pull/19075

### CODE REVIEW

File: apps/dav/lib/CardDAV/AddressBookRoot.php [hunk 1]
```diff
@@ -24,21 +24,28 @@
...
 	 * @param \Sabre\CardDAV\Backend\BackendInterface $carddavBackend
 	 * @param string $principalPrefix
 	 */
-	public function __construct(\Sabre\DAVACL\PrincipalBackend\BackendInterface $principalBackend, \Sabre\CardDAV\Backend\BackendInterface $carddavBackend, $principalPrefix = 'principals') {
+	public function __construct(\Sabre\DAVACL\PrincipalBackend\BackendInterface $principalBackend,
+								\Sabre\CardDAV\Backend\BackendInterface $carddavBackend,
+								PluginManager $pluginManager,
+								$principalPrefix = 'principals') {
 		parent::__construct($principalBackend, $carddavBackend, $principalPrefix);
-		$this->l10n = \OC::$server->getL10N('dav');
```
Comment (juliushaertl) [inline apps/dav/lib/CardDAV/AddressBookRoot.php]: Has an unused use and private property now.

Comment (ChristophWurst) [reply apps/dav/lib/CardDAV/AddressBookRoot.php]: Fixed :)

**Review by juliusknorr — APPROVED** [review]
Looks good otherwise and seems to do the trick with the parts already available in https://github.com/nextcloud/server/pull/19075

**Review by georgehrke — CHANGES_REQUESTED** [review]
`UserAddressbooks` should check for reserved names when creating a new collection.
Similar to here: https://github.com/nextcloud/server/blob/master/apps/dav/lib/CalDAV/CalendarHome.php#L76L84

**Comment by ChristophWurst** [comment]
Thanks @georgehrke. Added the check :)

**Review by rullzer — APPROVED** [review]
Didn't do extensive testing. But code looks sane

**Comment by ChristophWurst** [comment]
@georgehrke ok to merge this? :)

**Review by georgehrke — APPROVED** [review]
👍

**Comment by blizzz** [comment]
@ChristophWurst  the extension of info.xml has to be defined in https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/api/v1/release/info.xsd

**Comment by juliushaertl** [comment]
And https://github.com/nextcloud/server/blob/master/resources/app-info-shipped.xsd / https://github.com/nextcloud/server/blob/master/resources/app-info.xsd :wink:

**Comment by ChristophWurst** [comment]
:+1: I'll also add it for https://github.com/nextcloud/server/pull/19196 while at it

**Pull request merged**